### PR TITLE
ui/rotations: convert UserForm to typescript

### PR DESCRIPTION
**Pull request by nataliekorzh**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Converting the `UserForm` component to Typescript.

**Which issue(s) this PR fixes:**
Part of #2318

### CODE REVIEW

File: web/src/app/rotations/UserForm.tsx [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-import { PropTypes as p } from 'prop-types'
 import { FormContainer, FormField } from '../forms'
 import { UserSelect } from '../selection'
 
-export default function UserForm(props) {
+interface UserFormProps {
+  errors?: Array<Error>
+  onChange?: (value: { users: string[] }) => void
+  disabled?: boolean
+  value: { users?: Array<string> }
```
Comment (mastercactapus) [inline web/src/app/rotations/UserForm.tsx:9]: Myabe we should create a `UserFormValue` type to ensure these are identical

File: web/src/app/schedules/ScheduleRuleEditDialog.js [hunk 2]
```diff
@@ -32,51 +32,16 @@ const mutation = gql`
...
-  }
-
-  renderMutation(data, zone) {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {(commit, status) => this.renderDialog(data, commit, status, zone)}
-      </Mutation>
-    )
-  }
+export default function ScheduleRuleEditDialog({
```
Comment (mastercactapus) [inline web/src/app/schedules/ScheduleRuleEditDialog.js:35]: Was this intended to be included in this PR?

**Review by mastercactapus — CHANGES_REQUESTED** [review]

**Review by KatieMSB — CHANGES_REQUESTED** [review]

File: web/src/app/rotations/UserForm.tsx [hunk 3]
```diff
@@ -12,20 +18,11 @@ export default function UserForm(props) {
         fieldName='users'
         fullWidth
         label='Select User(s)'
-        multiple
+        multiline
```
[the remark this answers is not in the document]
Comment (nataliekorzh) [reply web/src/app/rotations/UserForm.tsx]: Agreed, I switched it to multiple

File: web/src/app/rotations/UserForm.tsx (hunk 1 above)
Comment (nataliekorzh) [reply web/src/app/rotations/UserForm.tsx]: Ok! Updated version with a separate `UserFormValue` is pushed

**Review by mastercactapus — APPROVED** [review]

**Pull request merged**